### PR TITLE
Cppparser initial

### PR DESCRIPTION
**Pull request by agladilin**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by agladilin** [comment]
@microsoft-github-policy-service agree

File: ports/cppparser/usage [hunk 1]
```diff
@@ -0,0 +1,4 @@
+cppparser provides CMake targets:
+
+find_package(cppparser CONFIG REQUIRED)
+target_link_libraries(main PRIVATE cppparser::cppparser)
```
Comment (dg0yt) [inline ports/cppparser/usage:4]:
```suggestion
  find_package(cppparser CONFIG REQUIRED)
  target_link_libraries(main PRIVATE cppparser::cppparser)
```

File: ports/cppparser/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,26 @@
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
```
Comment (dg0yt) [inline ports/cppparser/portfile.cmake:1]: Assuming the usual dllexport problem:
```suggestion
if(VCPKG_TARGET_IS_WINDOWS)
    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
endif()
```

File: ports/cppparser/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    REPO agladilin/cppparser
+    REF "${VERSION}"
+    SHA512 6fea5a6f7a172b8e828382871ad1dcb102ab700d617a1d2b753b6e0bfef7639e0b40778fdceb196e6b0b1b5e75c9173a1cb1eb667906ec7471118b74c5dbfce1
+    HEAD_REF master
+)
+
+
+vcpkg_cmake_configure(SOURCE_PATH "${SOURCE_PATH}")
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(PACKAGE_NAME "cppparser" CONFIG_PATH lib)
```
Comment (dg0yt) [inline ports/cppparser/portfile.cmake:14]:
```suggestion
vcpkg_cmake_config_fixup(CONFIG_PATH lib)
```
(Default.)

File: ports/cppparser/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+file(REMOVE_RECURSE
+    ${CURRENT_PACKAGES_DIR}/debug/include
```
Comment (dg0yt) [inline ports/cppparser/portfile.cmake:17]:
```suggestion
    "${CURRENT_PACKAGES_DIR}/debug/include"
```

File: ports/cppparser/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+)
+
+file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+configure_file("${CMAKE_CURRENT_LIST_DIR}/usage" "${CURRENT_PACKAGES_DIR}/share/${PORT}/usage" COPYONLY)
```
Comment (dg0yt) [inline ports/cppparser/portfile.cmake:21]:
```suggestion
file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
```

File: ports/cppparser/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+#install(
+#    FILES ${config_version_file} ${install_config}
+#    DESTINATION "${CPPPARSER_INSTALL_CONFIG_DIR}"
+#)
```
Comment (dg0yt) [inline ports/cppparser/portfile.cmake:26]:
```suggestion
```

File: ports/cppparser/vcpkg.json [hunk 7]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "cppparser",
+  "version": "0.1.0",
+  "description": "An easy, fast, and robust library to parse C/C++ source.",
+  "homepage": "https://github.com/agladilin/cppparser/tree/master",
```
Comment (dg0yt) [inline ports/cppparser/vcpkg.json:5]:
```suggestion
  "homepage": "https://github.com/agladilin/cppparser/",
```

File: ports/cppparser/usage (hunk 1 above)
Comment (agladilin) [reply ports/cppparser/usage]: done

File: ports/cppparser/portfile.cmake (hunk 2 above)
Comment (agladilin) [reply ports/cppparser/portfile.cmake]: done

File: ports/cppparser/portfile.cmake (hunk 3 above)
Comment (agladilin) [reply ports/cppparser/portfile.cmake]: done

File: ports/cppparser/portfile.cmake (hunk 4 above)
Comment (agladilin) [reply ports/cppparser/portfile.cmake]: done

File: ports/cppparser/portfile.cmake (hunk 6 above)
Comment (agladilin) [reply ports/cppparser/portfile.cmake]: done

File: ports/cppparser/vcpkg.json (hunk 7 above)
Comment (agladilin) [reply ports/cppparser/vcpkg.json]: done

**Comment by agladilin** [comment]
No reaction to my improvements. I'm going to start a new PR.

**Comment by dg0yt** [comment]
> No reaction to my improvements. I'm going to start a new PR.

Why? Note that the CI was still not "green".

**Pull request closed without merging**